### PR TITLE
[WIP] BinaryFormatter - Reenable sanity check

### PR DESCRIPTION
**Pull request by ViktorHofer**
1. DataSet and DataTable produced different blobs on Unix machines as their serialized payload contains line breaks: \r\n vs \n.
2. CookieContainer sets the machines domain to a field. Overwrote with reflection.
3. CompareInfo sets a SortVersion which is Windows OS machine dependent. Overwrote with reflection.
4. CookieCollection didn't contain a serializable field (m_version) probably because it was wrapped in pragmas and had its default value but only in CI, locally it was in the payload. The serialized value was also not 100% correct as the value should be the amount of cookies and not the default value of int. This isn't a breaking change as netfx doesn't use the value anymore for any operations. But with the change it's now always in the serialization payload.

### CODE REVIEW

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs [hunk 1]
```diff
@@ -412,16 +417,16 @@ public static IEnumerable<object[]> SerializableObjects_MemberData()
...
-
-            yield return new object[] { new SortVersion(12345, Guid.Parse("9fac6fbb-f683-42f5-a593-0d4a5e34b257")), new string[] { "AAEAAAD/////AQAAAAAAAAAEAQAAACBTeXN0ZW0uR2xvYmFsaXphdGlvbi5Tb3J0VmVyc2lvbgIAAAAMbV9ObHNWZXJzaW9uCG1fU29ydElkAAMIC1N5c3RlbS5HdWlkOTAAAAT+////C1N5c3RlbS5HdWlkCwAAAAJfYQJfYgJfYwJfZAJfZQJfZgJfZwJfaAJfaQJfagJfawAAAAAAAAAAAAAACAcHAgICAgICAgK7b6yfg/b1QqWTDUpeNLJXCw==", "AAEAAAD/////AQAAAAAAAAAEAQAAACBTeXN0ZW0uR2xvYmFsaXphdGlvbi5Tb3J0VmVyc2lvbgIAAAAMbV9ObHNWZXJzaW9uCG1fU29ydElkAAMIC1N5c3RlbS5HdWlkOTAAAAT+////C1N5c3RlbS5HdWlkCwAAAAJfYQJfYgJfYwJfZAJfZQJfZgJfZwJfaAJfaQJfagJfawAAAAAAAAAAAAAACAcHAgICAgICAgK7b6yfg/b1QqWTDUpeNLJXCw==" } };
+            var sortVersion = new SortVersion(12345, Guid.Parse("9fac6fbb-f683-42f5-a593-0d4a5e34b257"));
+            var compareInfo = new CultureInfo("").CompareInfo;
+            // We set the sort version (if the field exists as it is optional) explicitely so that we aren't
```
Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: typo, explicitly

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs [hunk 2]
```diff
@@ -417,16 +417,16 @@ public static IEnumerable<object[]> SerializableObjects_MemberData()
...
-
-            yield return new object[] { new SortVersion(12345, Guid.Parse("9fac6fbb-f683-42f5-a593-0d4a5e34b257")), new string[] { "AAEAAAD/////AQAAAAAAAAAEAQAAACBTeXN0ZW0uR2xvYmFsaXphdGlvbi5Tb3J0VmVyc2lvbgIAAAAMbV9ObHNWZXJzaW9uCG1fU29ydElkAAMIC1N5c3RlbS5HdWlkOTAAAAT+////C1N5c3RlbS5HdWlkCwAAAAJfYQJfYgJfYwJfZAJfZQJfZgJfZwJfaAJfaQJfagJfawAAAAAAAAAAAAAACAcHAgICAgICAgK7b6yfg/b1QqWTDUpeNLJXCw==", "AAEAAAD/////AQAAAAAAAAAEAQAAACBTeXN0ZW0uR2xvYmFsaXphdGlvbi5Tb3J0VmVyc2lvbgIAAAAMbV9ObHNWZXJzaW9uCG1fU29ydElkAAMIC1N5c3RlbS5HdWlkOTAAAAT+////C1N5c3RlbS5HdWlkCwAAAAJfYQJfYgJfYwJfZAJfZQJfZgJfZwJfaAJfaQJfagJfawAAAAAAAAAAAAAACAcHAgICAgICAgK7b6yfg/b1QqWTDUpeNLJXCw==" } };
+            var sortVersion = new SortVersion(12345, Guid.Parse("9fac6fbb-f683-42f5-a593-0d4a5e34b257"));
+            var compareInfo = new CultureInfo("").CompareInfo;
+            // We set the sort version (if the field exists as it is optional) explicitily so that we aren't
```
Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: typo explicitly

Comment (ViktorHofer) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: ooh my. thanks ;)

**Comment by ViktorHofer** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by ViktorHofer** [comment]
I will fix the typo. @danmosemsft please sign off

**Pull request merged**